docs/ref/forms/api.txt

```diff
@@ -830,107 +830,228 @@ The field-specific output honors the form object's ``auto_id`` setting::
     >>> print(f['message'])
     <input type="text" name="message" id="id_message" />
 
-For a field's list of errors, access the field's ``errors`` attribute.
+Attributes of ``BoundField``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. attribute:: BoundField.errors
+.. attribute:: BoundField.auto_id
 
-A list-like object that is displayed as an HTML ``<ul class="errorlist">``
-when printed::
+    Calculates and returns the ID attribute for this BoundField if the
+    associated Form has specified auto_id. Returns an empty string otherwise.
 
-    >>> data = {'subject': 'hi', 'message': '', 'sender': '', 'cc_myself': ''}
-    >>> f = ContactForm(data, auto_id=False)
-    >>> print(f['message'])
-    <input type="text" name="message" />
-    >>> f['message'].errors
-    ['This field is required.']
-    >>> print(f['message'].errors)
-    <ul class="errorlist"><li>This field is required.</li></ul>
-    >>> f['subject'].errors
-    []
-    >>> print(f['subject'].errors)
+    .. note::
+       This attribute gets calculated by the form only, so if the field's
+       widget overrides :meth:`~django.forms.Widget.id_for_label` this ID
+       is likely to be wrong.
```

Comment:
I agree that `as_text()` and `as_textarea()` are not needed so I will remove them from the docs.

There is a use case for `as_widget()` though. In one of my projects I have a template filter that is called `add_class` and I use it like this:
```html+django
{{ my_form.my_field|add_class:'class-added-to-input-field' }}
```
It uses the `attrs` argument to `as_widget()` to easily add the class attribute. That wouldn't be possible like that if the code was just private inside `__str__()`.
Although this use case may be invalidated by the upcoming template based widget rendering.

I'm not sure about `as_hidden()`. I personally never used it, since you can easily emulate it with `html_name` but I'm sure there are a few people using it.